query_builder/advanced.py

```diff
@@ -122,8 +123,16 @@ def handle_join(self, table, query, config):
 
     def get_select_query(self, table, query, config):
         if 'groupby' in config:
-            query = self.get_groupby_query(table, query, config.get('groupby'))
-        if 'having' in config:
+            if config.get('selectall'):
+                raise LookupError('Columns must be explicitly SELECTED for queries that use GROUP BY clauses')
+            if 'columns' in config:
+                query = self.get_groupby_query(table, query, config.get('groupby'), config.get('columns'))
+                if 'having' in config:
+                    query = self.get_having_query(table, query, config.get('having'))
+            else:
+                raise LookupError('Columns must be explicitly SELECTED for queries that use GROUP BY clauses')
+
+        if 'having' in config and 'groupby' not in config:
```

Comment:
This should not happen! If it does, return an error ... `HAVING` is to `GROUP BY` as `WHERE` is to `SELECT` ... you can't have one without the other.
I don't think it should appear in this function at all ... it should be in the `get_groupby_query` function

Reply:
Thinking about it, this check is not needed even in the `get_groupby_query` function, because the frontend won't allow a having clause without a `groupby`. And by the `get_groupby_query` function is called only when there is a `groupby` in the config